src/util/graph.h

```diff
@@ -458,6 +458,34 @@ void grapht<N>::visit_reachable(node_indext src)
     nodes[index].visited = true;
 }
 
+/// Add to `set` nodes that are reachable from `set`
+/// \param set: set of source nodes, must be a container with an
+///   `insert(const value_type&)` method.
+/// \param for_each_successor: function which given a node `n` and a function
+///   `f`, applies `f` on all successors of `n`.
+template <class Container, typename nodet = typename Container::value_type>
+void get_reachable(
+  Container &set,
+  const std::function<void(
+    const typename Container::value_type &,
+    const std::function<void(const typename Container::value_type &)> &)>
+    &for_each_successor)
+{
+  std::vector<nodet> stack;
```

Comment:
Don't store `nodet`, use `node_indext`. Going from a `node_indext` to a `nodet` is constant time, but a `node_indext` will typically consume a lot less memory.

Reply:
Since this function is generic you can use it different ways, in particular if you give a set of node_indext then the algorithm will only manipulate node_indext. nodet here does not have to correspond to graph::node_t.

Reply:
Got it.